test/threads.jl

```diff
@@ -141,3 +141,41 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic fences
+type CommBuf
+    barrier::Atomic{Int}
+    data::Vector{Float64}
+    correct::Bool
+    CommBuf() = new(Atomic{Int}(0), Vector{Float64}(100), false)
+end
+function test_fence_write(commbuf::CommBuf)
+    sleep(0.1)
+    for i in eachindex(commbuf.data)
+        commbuf.data[i] = i
+    end
+    atomic_fence()
+    commbuf.barrier[] = 1
```

Comment:
P.S. I think the following sentence on the GCC wiki is confusing,

> Since 'y' is not an atomic variable, the store to 'y' happens-before the store to 'x'

I thnk the only thing that matters is that the two happens on the same thread and the second one is an atomic release store. Whether `y` is atomic or not doesn't matter here (and making it atomic shouldn't make the assertion fail.....).